src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -558,15 +571,39 @@ private[fastq] object FastqDemultiplexer {
   /** Stores the minimal information for a single template read. */
   case class DemuxRecord(name: String, bases: String, quals: String, molecularBarcode: Option[String],
                          sampleBarcode: Option[String], readNumber: Int, pairedEnd: Boolean, comment: Option[String],
-                         originalBases: Option[String] = None, originalQuals: Option[String] = None)
+                         originalBases: Option[String] = None, originalQuals: Option[String] = None) {
+
+    /** Masks bases that have a quality score less than or equal to a specified threshold.
+      * @param threshold The threshold for masking bases, exlusive. Bases with a quality score < threshold will be masked
+      * @param qualityEncoding The encoding used for quality scores in the Fastq file.
+      * @return a new DemuxRecord with updated bases
+      */
+    def maskLowQualityBases(threshold: PhredScore, qualityEncoding: QualityEncoding): DemuxRecord = {
+      if (threshold <= 0 || this.quals.forall(_ > threshold)) this else {
```

Comment:
bug in my previous suggestion
```suggestion
      if (threshold <= 0 || this.quals.forall(_ >= threshold)) this else {
```